File: backend/src/Entity/ChatMessage.php
<?php

declare(strict_types=1);

namespace App\Entity;

use App\Repository\ChatMessageRepository;
use Doctrine\ORM\Mapping as ORM;

#[ORM\Entity(repositoryClass: ChatMessageRepository::class)]
#[ORM\Table(name: 'chat_messages')]
class ChatMessage
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column(type: 'bigint')]
    private ?int $id = null;

    #[ORM\ManyToOne(targetEntity: ChatConversation::class, inversedBy: 'messages')]
    #[ORM\JoinColumn(nullable: false, onDelete: 'CASCADE')]
    private ChatConversation $conversation;

    #[ORM\ManyToOne(targetEntity: User::class)]
    #[ORM\JoinColumn(nullable: false, onDelete: 'CASCADE')]
    private User $sender;

    #[ORM\Column(type: 'text')]
    private string $content;

    #[ORM\Column(type: 'datetime')]
    private \DateTimeInterface $createdAt;

    public function __construct()
    {
        $this->createdAt = new \DateTime();
    }

    public function getId(): ?int
    {
        return $this->id !== null ? (int) $this->id : null;
    }

    public function getConversation(): ChatConversation
    {
        return $this->conversation;
    }

    public function setConversation(ChatConversation $conversation): static
    {
        $this->conversation = $conversation;
        return $this;
    }

    public function getSender(): User
    {
        return $this->sender;
    }

    public function setSender(User $sender): static
    {
        $this->sender = $sender;
        return $this;
    }

    public function getContent(): string
    {
        return $this->content;
    }

    public function setContent(string $content): static
    {
        $this->content = $content;
        return $this;
    }

    public function getCreatedAt(): \DateTimeInterface
    {
        return $this->createdAt;
    }

    public function setCreatedAt(\DateTimeInterface $createdAt): static
    {
        $this->createdAt = $createdAt;
        return $this;
    }
}

File: mobile/lib/core/models/chat_message.dart
class ChatMessage {
  final int id;
  final int conversationId;
  final int senderId;
  final String content;
  final DateTime createdAt;

  const ChatMessage({
    required this.id,
    required this.conversationId,
    required this.senderId,
    required this.content,
    required this.createdAt,
  });

  factory ChatMessage.fromJson(Map<String, dynamic> j) {
    final int senderId;
    if (j['senderId'] is int) {
      senderId = j['senderId'] as int;
    } else if (j['sender'] is Map<String, dynamic>) {
      senderId = (j['sender'] as Map<String, dynamic>)['id'] as int;
    } else {
      senderId = 0;
    }
    return ChatMessage(
      id: (j['id'] as int?) ?? 0,
      conversationId: j['conversationId'] as int,
      senderId: senderId,
      content: j['content'] as String,
      createdAt: DateTime.parse(j['createdAt'] as String),
    );
  }
}
